src/java/org/apache/cassandra/utils/Throwables.java

```diff
@@ -288,12 +288,17 @@ public static Throwable unwrapped(Throwable t)
                unwrapped instanceof InvocationTargetException)
             unwrapped = unwrapped.getCause();
 
-        // I don't think it make sense for those 2 exception classes to ever be used with null causes, but no point
-        // in failing here if this happen. We still wrap the original exception if that happen so we get a sign
-        // that the assumption of this method is wrong.
-        return unwrapped == null
-               ? new RuntimeException("Got wrapping exception not wrapping anything", t)
-               : unwrapped;
+        if (unwrapped == null)
+            throw new RuntimeException("Got wrapping exception not wrapping anything", t);
+
+        if (!t.getStackTrace()[t.getStackTrace().length - 1].getClassName().equals(Thread.class.getName()))
```

Comment:
I don't follow this logic, can you add some documentation for what this is trying to do?

Reply:
Shame on me but I probably accidentally committed some junk. I removed that condition, sorry for the confusion.